### PR TITLE
Various optimizations

### PR DESCRIPTION
**Pull request by fabrichter**

### CODE REVIEW

File: languagetool-core/src/main/java/org/languagetool/Language.java [hunk 1]
```diff
@@ -93,6 +100,12 @@ public abstract class Language {
   private Chunker postDisambiguationChunker;
   private Synthesizer synthesizer;
 
+  private final String shortCodeWithCountryAndVariant;
+
+  protected Language() {
```
Comment (SteVio89) [inline languagetool-core/src/main/java/org/languagetool/Language.java:105]: This is not used by -> shortCodeWithCountryAndVariant is always NULL

File: languagetool-core/src/main/java/org/languagetool/Language.java [hunk 2]
```diff
@@ -631,6 +644,10 @@ public final String getTranslatedName(ResourceBundle messages) {
    * @since 3.6
    */
   public final String getShortCodeWithCountryAndVariant() {
+    return shortCodeWithCountryAndVariant;
```
Comment (SteVio89) [inline languagetool-core/src/main/java/org/languagetool/Language.java:647]: Returns NULL

File: languagetool-core/src/main/java/org/languagetool/Language.java [hunk 3]
```diff
@@ -926,7 +947,7 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return getShortCodeWithCountryAndVariant().hashCode();
+    return shortCodeWithCountryAndVariant.hashCode();
```
Comment (SteVio89) [inline languagetool-core/src/main/java/org/languagetool/Language.java:950]: NullPointerException

File: languagetool-core/src/main/java/org/languagetool/Language.java (hunk 1 above)
Comment (fabrichter) [reply languagetool-core/src/main/java/org/languagetool/Language.java:105]: this is impossible, the field is final and not initializing it (by adding another constructor) will cause a compiler error
and in subclasses, superclass constructors are always called, either explicit or implicit

File: languagetool-core/src/main/java/org/languagetool/Language.java (hunk 2 above)
Comment (fabrichter) [reply languagetool-core/src/main/java/org/languagetool/Language.java:647]: see comment above

**Pull request merged**